pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -180,15 +180,17 @@ func parseHostname(hostname model.Hostname) (name string, namespace string, err
 	return
 }
 
+var protocolGRPCWebLowercase = strings.ToLower(string(model.ProtocolGRPCWeb))
+
 // ConvertProtocol from k8s protocol and port name
 func ConvertProtocol(name string, proto v1.Protocol) model.Protocol {
 	out := model.ProtocolTCP
 	switch proto {
 	case v1.ProtocolUDP:
 		out = model.ProtocolUDP
 	case v1.ProtocolTCP:
-		prefix := name
-		if strings.HasPrefix(strings.ToLower(prefix), strings.ToLower(string(model.ProtocolGRPCWeb))) {
+		prefix := strings.ToLower(name)
+		if strings.HasPrefix(name, protocolGRPCWebLowercase) {
```

Comment:
This optimization looks fine. However, I can think of even better idea I think https://play.golang.org/p/GWkmWaRrC5r.

Reply:
Nice. I was previously looking for a case-insensitive comparison, but I missed this.